experimental_official_dockerfiles/master/Dockerfile

```diff
@@ -0,0 +1,61 @@
+################################################################################
+FROM ubuntu:20.04 as builder
+################################################################################
+
+# Install devtoolset build dependencies
+COPY setup.packages.sh setup.packages.sh
+COPY builder.packages.txt builder.packages.txt
+RUN /setup.packages.sh /builder.packages.txt
+
+# Install devtoolset-7 in /dt7 with gclibc 2.12 and libstdc++ 4.4, for building
+# manylinux2010-compatible packages. Scripts expect to be in the root directory
+COPY devtoolset/fixlinks.sh /fixlinks.sh
+COPY devtoolset/rpm-patch.sh /rpm-patch.sh
+COPY devtoolset/build_devtoolset.sh /build_devtoolset.sh
+RUN /build_devtoolset.sh devtoolset-7 /dt7 
+
+################################################################################
+FROM nvidia/cuda:11.2.1-base-ubuntu20.04 as devel
+################################################################################
+
+COPY --from=builder /dt7 /dt7
+
+# Install required development packages but delete unneeded CUDA bloat
+# CUDA must be cleaned up in the same command to prevent Docker layer bloating
+COPY setup.sources.sh /setup.sources.sh
+COPY setup.packages.sh /setup.packages.sh
+COPY setup.cuda.sh /setup.cuda.sh
+COPY devel.packages.txt /devel.packages.txt
+RUN /setup.sources.sh && /setup.packages.sh /devel.packages.txt && /setup.cuda.sh
+
+# Install various tools.
+# - bats: bash unit testing framework
+# - bazelisk: always use the correct bazel version
+# - buildifier: clean bazel build deps
+# - buildozer: clean bazel build deps
+RUN git clone https://github.com/bats-core/bats-core.git && bats-core/install.sh /usr/local
```

Comment:
Do we need to maintain the checkout in the layer?

Reply:
No, I've deleted it now, and also chosen a specific tag.